libs/common/src/sql-db-entities/helpers/user-id.helper.ts

```diff
@@ -0,0 +1,8 @@
+type RequestSource = { platform: 'PORTAL' | 'TFM'; userId: string } | { platform: 'SYSTEM' };
+
+export type DbRequestSourceParam = {
+  requestSource: RequestSource;
+};
+
+export const getDbUpdatedByUserId = (requestSource: RequestSource) =>
```

Comment:
Could we maybe create a type for `UpdatedByUserId` for the sake of some documentation? It already took me a minute to figure out why this will be different to `uploadedByUserId`, and I think we can extract the type from `RequestSource`, although I also think it would involve some non-trivial typescript. I don't feel too strongly about what kind of documentation we have for it, but I do think it may benefit from some

Reply:
Have add a new `DbAuditUpdatedByUserId` type. I could have gone further and add a Platform type like this
```
type Platform = {
  PORTAL: 'PORTAL',
  TFM: 'TFM',
  SYSTEM: 'SYSTEM',
}
```
and used that where required, but I think that overcomplicated things and obscures the intention. If any of the values between `DbAuditUpdatedByUserId`, `RequestSource` and `getDbAuditUpdatedByUserId` go our of sync then a type error is thrown, so I think we're safe.

Have also renamed the function to `getDbAuditUpdatedByUserId` and added a JSDoc comment to hopefully make the use clearer.

Let me know what you think.

Reply:
You could just define the `Platform` type with
```typescript
type Platform = RequestSource['platform'];
```
and get all three of the platforms if you wanted to, but that would still mean your `DbAuditUpdatedByUserId` type would end up looking like
```typescript
type DbAuditUpdatedByUserId = {
    [P in Platform]: P extends 'SYSTEM' ? P : `${P}-${UserId}`;
}[Platform];
```
which I think it too complex for this project. I'm happy with the current implementation you have come up with, so feel free to resolve this comment :)